internal/app/secretless/proxy.go

```diff
@@ -59,6 +68,30 @@ func (p *Proxy) Listen(listenerConfig config.Listener) plugin_v1.Listener {
 		RunHandlerFunc: p.RunHandlerFunc,
 	}
 
+	if listenerConfig.Protocol == "pg" {
```

Comment:
This pg stuff seems completely independent from the loadkeys change

Reply:
Also, this seems like it should go into a function rather than being inlined into `Listen`